pkg/tools/flags/url_slice_value.go

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flags
+
+import (
+	"bytes"
+	"encoding/csv"
+	"net/url"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/pflag"
+)
+
+const (
+	urlSliceType = "url-slice"
+)
+
+type urlSliceValue struct {
+	values  *[]*url.URL
+	changed bool
+}
+
+// NewURLSliceValue - create new value for URLs
+func NewURLSliceValue(p *[]*url.URL) pflag.Value {
+	v := &urlSliceValue{
+		values:  p,
+		changed: false,
+	}
+	*p = []*url.URL{}
+	return v
+}
+
+func readAsCSV(val string) ([]*url.URL, error) {
```

Comment:
Interesting... by reading as csv you basically make both the apparent unitary case:
unix://foo
And the multiple case
unix://foo,10.10.10.1:9111 valid.  Correct?

Reply:
Yes.